toolchain/sem_ir/typed_insts.h

```diff
@@ -444,6 +444,25 @@ struct FacetTypeAccess {
   InstId facet_id;
 };
 
+struct FloatLiteral {
+  static constexpr auto Kind =
+      InstKind::FloatLiteral.Define<Parse::RealLiteralId>("float_literal");
+
+  TypeId type_id;
+  FloatId float_id;
+};
+
+struct FloatType {
+  static constexpr auto Kind =
+      InstKind::FloatType.Define<Parse::NodeId>("float_type");
+
+  TypeId type_id;
+  FloatKind float_kind;
```

Comment:
Do you anticipate a use for this, or is this just here to match `IntType`? I think we can probably do without this `kind` value for now.

(The bit width might not be quite enough to distinguish, e.g. a 128-bit extended IEEE format from PPC double-double, or to distinguish between different 16-bit FP formats, but nor can our `fN` notation. I think we can defer worrying about this until we have a design for a wider set of floating point types.)

Reply:
That was me following `IntType` and not realizing I didn't end up using the field 😅 . Just removed it.

Thanks for pointing out the bit width as well! Let's defer it for now like you said. I'd be happy to work on that once the design we need is in place.